### PR TITLE
docs: update README

### PR DESCRIPTION
**Pull request by jsmassa**
Changes:
- Update Changelog
- Separate usage examples into execution vs. store type
- rename `new-fs-store` to `connect-fs-store`

### CODE REVIEW

File: README.org [hunk 1]
```diff
@@ -70,18 +84,46 @@ is not critical for you (e.g. for a session store). Note that the database will
...
+
 *** IndexedDB
     :PROPERTIES:
     :CUSTOM_ID: h:ccbb272e-24b1-4f1e-b525-dd07c4e0e9b4
     :END:
 
+[[https://developer.mozilla.org/en-US/docs/IndexedDB][IndexedDB]] is provided as reference implementation for
+ClojureScript browser backends.
+
 For IndexedDB there is no internal JSON-representation of the underlying
```
Comment (whilo) [inline README.org:107]: This is not true anymore, we actually use the `fress` library from @pkpkpk and he updated the backend.

Comment (whilo) [reply README.org:107]: So this paragraph can be removed.

**Review by whilo — CHANGES_REQUESTED** [review]
It would be good to also mention the `keys` function and the fact that there is a garbage collector and how it works: It gets a whitelist and deletes the difference between an invocation of `keys` and this whitelist. Also all backends except for JDBC should be declared deprecated as they are not compatible with the current version of konserve anymore.

File: README.org (hunk 1 above)
Comment (jsmassa) [reply README.org:107]: Done.

**Comment by jsmassa** [comment]
I added a tiny section about the garbage collector. I wouldn't know what to write about the `keys` function though, seems more like a detail to me. Do we have a cljdoc for konserve's API namespaces?

**Comment by jsmassa** [comment]
I also updated the paragraph about the backends.

**Review by whilo — DISMISSED** [review]
I cleaned up a bit more, please check. Otherwise it looks good.

File: README.org [hunk 2]
```diff
@@ -183,182 +296,32 @@ for records, if you need them.
...
-    konserve.js.assoc_in(store, ["foo"], 42, function(v) {});
-    konserve.js.get_in(store,
-                       ["foo"],
-                       function(v) { console.log(v); });
-    konserve.js.update_in(store,
-                          ["foo"],
-                          function(v) { return v+1; },
-                          function(res) { console.log("Result:", res); });
-#+END_SRC
+   We provide a [[file:doc/backend.org][backend implementation guide]] .
```
Comment (jsmassa) [inline README.org:304]: Space before dot

File: README.org [hunk 3]
```diff
@@ -183,182 +296,32 @@ for records, if you need them.
...
 
 ** TODO
```
Comment (jsmassa) [inline README.org:306]: For some reason this headline doesn't get shown when I look at the rendered version on GitHub. Any ideas why?

Comment (whilo) [reply README.org]: Because capitalized TODO is a keyword in org-mode to annotate headlines as a todo item. I changed it.

File: README.org (hunk 2 above)
Comment (whilo) [reply README.org]: Fixed.

**Review by whilo — APPROVED** [review]

**Pull request merged**